checks.d/openstack.py

```diff
@@ -512,3 +650,57 @@ def check(self, instance):
         except IncompleteConfig:
             self.warning("Configuration Incomplete! Check your openstack.yaml file")
             return
+
+
+    #### Local Info accessors
+    def get_local_hypervisor(self):
+        # Look up hypervisors available filtered by my hostname
+        host = self.get_my_hostname()
+        hyp = self.get_all_hypervisor_ids(filter_by_host=host)
+        return hyp[0]
+
+    def get_local_project(self):
```

Comment:
Hm, interesting. `local_only` is a very coarse-grained option right now, a little bit by design - so that  existing users (such as yourself :)) can continue to get the old functionality just by flipping a single flag. I did not account for the possibility that `local_only` and a `check_all_*` option would be run in parallel.

Just thinking out loud about ways we can support your use case. In this version, even though a hypervisor can be involved in many projects, the agent is only scoped to a single project. This is kind of a glaring limitation and means that you would need to run one agent per unique `(hypervisor, project)` to get complete coverage of your project metrics - in the process duplicating the hypervisor-level metrics. I think the way to solve this is to have different project domains listed as `instances` in the check, rather than under `init_config`. This allows an agent running on a single host to perform `identity::get_project` on multiple projects, and allows you to deploy one agent per hypervisor in `local_only` mode without issue.

The above change should probably be made in a separate PR. For now, it looks like the best we can do is have one agent running `check_all_projects`, and a separate agent in `local_only` mode :(

Sorry for this, but you have pointed out a big improvement that should be made to the check